src/control/lib/hardware/sysfs/provider.go

```diff
@@ -156,44 +139,52 @@ func (s *Provider) addPCIDevices(topo *hardware.Topology, subsystem string) erro
 		}
 
 		if isNetvscDevice(path, subsystem) {
+			netvscPaths = append(netvscPaths, path)
 			return nil
 		}
 
-		var dev *hardware.PCIDevice
-		switch {
-		case isNetwork(subsystem):
-			dev, err = s.getNetworkDevice(path, subsystem)
-			if err != nil {
-				s.log.Tracef(err.Error())
-				return nil
-			}
-		default:
-			return nil
-		}
+		return s.addPCIDevice(topo, subsystem, path)
+	})
 
-		numaID, err := s.getNUMANode(path)
-		if err != nil {
-			s.log.Tracef("using default NUMA node, unable to get: %s", err.Error())
-			numaID = 0
-		}
+	if err != io.EOF && err != nil {
+		return err
+	}
 
-		pciAddr, err := s.getPCIAddress(path)
-		if err != nil {
-			s.log.Debug(err.Error())
-			return nil
+	for _, path := range netvscPaths {
+		if err := s.addNetvscDevice(topo, path); err != nil {
+			return err
 		}
-		dev.PCIAddr = *pciAddr
+	}
 
-		s.log.Tracef("adding device found at %q (type %s, NUMA node %d)", path, dev.Type, numaID)
+	return nil
+}
 
-		return topo.AddDevice(uint(numaID), dev)
-	})
+func (s *Provider) addPCIDevice(topo *hardware.Topology, subsystem string, path string) error {
+	if !isNetwork(subsystem) {
+		return nil
+	}
 
-	if err == io.EOF || err == nil {
+	dev, err := s.getNetworkDevice(path, subsystem)
+	if err != nil {
+		s.log.Tracef(err.Error())
 		return nil
 	}
 
-	return err
+	numaID, err := s.getNUMANode(path)
+	if err != nil {
+		s.log.Tracef("using default NUMA node, unable to get: %s", err.Error())
```

Comment:
is this a normal occurrence for getNUMANode to fail? shouldn't we log to something more than trace as this will be ignored in most cases